bundles/org.eclipse.equinox.p2.core/src/org/eclipse/equinox/p2/core/IProvisioningAgent.java

```diff
@@ -165,4 +165,32 @@ default boolean getBooleanProperty(String key, boolean defaultValue) {
 		return Boolean.parseBoolean(getProperty(key, Boolean.toString(defaultValue)));
 	}
 
+	/**
+	 * Returns an agent bound property as an int, the default implementation
+	 * delegates to {@link IProvisioningAgent#getIntProperty(String, int)} with
+	 * <code>0</code> as the default.
+	 *
+	 * @since 2.13
+	 */
+	default int getIntProperty(String key) {
+		return getIntProperty(key, 0);
+	}
+
+	/**
+	 * Returns an agent bound property as an int, the default implementation
+	 * delegates to {@link IProvisioningAgent#getProperty(String)} with the given
+	 * default as a String and parses the result as by
+	 * {@link Boolean#parseBoolean(String)}.
+	 *
+	 * @since 2.13
+	 */
+	default int getIntProperty(String key, int defaultValue) {
+		String propertyValue = getProperty(key, Integer.toString(defaultValue));
```

Comment:
Also a nit, but I suppose one could call getProperty without a default value and if the propertyValue comes back null return the default. I don't mind either way, so don't feel compelled to change it.

Reply:
In that case, `Integer.parseInt` would throw a NPE leading to the default value being returned.

Reply:
It's not wrong how it is, but if you checked  getProperty(key) for returning null, you'd not have to convert defaultValue to a string and then parse that string.  As I said it's really a nit.

Reply:
Enough nit picking.  It's okay like this.